tests/test_downloadermiddleware_httpproxy.py

```diff
@@ -123,7 +123,11 @@ def test_proxy_already_seted(self):
 
     def test_no_proxy(self):
         os.environ['http_proxy'] = 'https://proxy.for.http:3128'
+        os.environ['no_proxy'] = '/var/run/docker.sock'
         mw = HttpProxyMiddleware()
+        # '/var/run/docker.sock' may be used by the user for
+        # no_proxy value but is not parseable and should be skipped
+        assert 'no' not in mw.proxies
```

Comment:
What would you think about creating a separate method just to test this, instead of modifying an existing test method?

Reply:
Yeah that makes sense.
Also should I do anything about the multi-valued no_proxy env variable like `no_proxy=127.0.0.1,example.com,another.example.com`? `_get_proxy`  does not deal with the multiple values well.

Reply:
> Also should I do anything about the multi-valued no_proxy env variable like no_proxy=127.0.0.1,example.com,another.example.com? _get_proxy does not deal with the multiple values well.

Since it is not necessary to fix the source issue, and it may complicate the review, I would just go ahead with the current change.

I suggest that you instead simply open an issue about it, so that we do not loose track about it. And, only if you feel motivated, feel free to work on it yourself.